packages/ui/src/kitchenSinkItem.tsx

```diff
@@ -60,22 +71,46 @@ export class KitchenSinkItem {
     }
 
     public isActive(): boolean {
-        return this._options.isExpanded(this._options.data.position);
+        return this._options.isActive(this);
     }
 
     public activate(): void {
-        this._options.expand(this._options.data.position, this._options.data.expandMode);
+        this._options.activate(this);
+    }
+
+    private _activationStateChange = ({ item }: ItemActiveStateChangeEvent) => {
+        // handle only if relevant to this item
+        if (this === item) {
+            this.update();
+        }
+    };
+
+    public deactivate(): void {
+        this._options.deactivate(this);
```

Comment:
since you expose `destroy` mechanism, you are responsible also not to handle any public api provided. so this method and other public method should start with something like
```
if (this._destroyed) {
   logger.warn(`cannot perform 'deactivate' action, item is marked as destroyed`, {});
   return;
}

Reply:
done

Reply:
cool, actually you followed my suggestion exactly which caused to duplicated code :/ 
I meant something like. the more proper way is using a shared method like following
```
   private _isDestroyed() : boolean {
 if (this._destroyed) {
            this._logger.warn(`cannot perform requested call, item was marked as destroyed`, {});
            return true;
        }

return false;
}
```

then in any method just do
```
    public deactivate() {
         if (this._isDestroyed()) {
             return;
        } 
    }
```